zip/src/main/kotlin/io/github/paulgriffith/kindling/zip/ZipView.kt

```diff
@@ -143,83 +130,53 @@ class ZipView(path: Path) : ToolPanel() {
                         openIndividually.name = "Open ${selectedPaths.size} files individually"
                     }
                     add(openIndividually)
-                }
-            }
-        }
 
-        add(backupInfo, "north, pad 6 6 6 6")
-        add(sidebar, "west, width 200::300, pad 6 6 6 6")
-        add(tabbedPane, "dock center")
-    }
-
-    data class ZipFile(val path: Path, val attributes: BasicFileAttributes)
-
-    class TabWrapper(
-        val attributes: List<ZipFile>,
-        val pathView: PathView,
-    ) : JPanel(MigLayout("ins 6")) {
-        private val path = attributes.singleOrNull()?.path
-
-        private val saveAs = JButton("Save As").apply {
-            addActionListener {
-                exportFileChooser.selectedFile = File(path?.last().toString())
-                if (exportFileChooser.showSaveDialog(this@TabWrapper) == JFileChooser.APPROVE_OPTION) {
-                    pathView.provider.newInputStream(path).use { file ->
-                        exportFileChooser.selectedFile.toPath().outputStream().use(file::copyTo)
+                    if (selectedPaths.size == 1) {
+                        add(
+                            Action("Save As") {
+                                val selectedNode = selectedPaths.first().lastPathComponent as PathNode
+                                exportFileChooser.selectedFile = File(selectedNode.userObject.name)
+                                if (exportFileChooser.showSaveDialog(this@attachPopupMenu) == JFileChooser.APPROVE_OPTION) {
+                                    provider.newInputStream(path).use { file ->
+                                        exportFileChooser.selectedFile.toPath().outputStream().use(file::copyTo)
+                                    }
+                                }
+                            },
+                        )
```

Comment:
Moving this to the context menu saves some screen real estate; I dropped the "header" label as well to make things a bit more content-dense